src/thunderbots/software/ai/navigator/obstacle/obstacle.h

```diff
@@ -3,27 +3,82 @@
  */
 #pragma once
 
+#include "ai/world/ball.h"
 #include "ai/world/robot.h"
+#include "geom/angle.h"
 #include "geom/point.h"
 #include "geom/polygon.h"
 #include "geom/util.h"
+#include "shared/constants.h"
 
 class Obstacle
 {
    public:
+    static Obstacle createPlaceholderObstacle();
+
+    static Obstacle createRobotObstacle(const Robot& robot, bool enable_velocity_cushion);
+
     /*
      * Gets the boundary polygon around the given robot obstacle that other robots
-     * should not enter with a buffer scaled by radiusScaling and
-     * velocity_projection_scaling
+     * should not enter with a buffer scaled by radius_scaling and
+     * velocity_cushion_scaling
      *
      * @param robot robot to create obstacle boundary polygon around
-     * @param radiusScaling safety factor to vary the radius size of the buffer zone
-     * (1=default), must be greater than 0
-     * @param velocity_projection_scaling scales the projection buffer in the direction of
-     * the velocity (1=default), must be greater than 0
+     * @param radius_scaling multiplicatively scales the radius of the obstacle
+     * @param velocity_cushion_scaling multiplicatively scales the cushion of
+     * velocity of the robot
+     * @param enable_velocity_cushion if true, enables the cushion of the velocity
+     * of the robot as component of the obstacle
+     *
+     * @note the radius is by default the minimum distance needed for two robots to clear
+     * each other
+     * @note the velocity cushion is by default the distance that can be travelled in
+     * one tick
```

Comment:
For more detailed explanations, generally these would go after the short description at the start. As an additional note, technically the first section of the javadoc is only one line (but this doesn't matter too much for us). With some tweaks, this javadoc comment could look like:
```
    /*
     * Creates a Boundry Polygon for a given Robot
     *
     * The radius is by default the minimum distance needed for two robots to clear
     * each other
     * The velocity cushion is by default the distance that can be travelled in
     * one tick
     *
     * @param robot robot to create obstacle boundary polygon around
     * @param radius_scaling multiplicatively scales the radius of the obstacle
     * @param velocity_cushion_scaling multiplicatively scales the cushion of
     * velocity of the robot
     * @param enable_velocity_cushion if true, enables the cushion of the velocity
     * of the robot as component of the obstacle
     *
     *
     * @return a six-sided Polygon to represent the boundary around the obstacle
     */
```

Reply:
It doesn't look like we're setting default values for either of these things though? Even if we were, the fact that they have default values is generally pretty self-evident from looking at the function signature.

Reply:
I've added more comments